model/connect/MySQLStatement.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/**
+ * Provides a record-view for mysqli statements
+ *
+ * By default streams unbuffered data, but seek(), rewind(), or numRecords() will force the statement to
+ * buffer itself and sacrifice any potential performance benefit.
+ */
+class MySQLStatement extends SS_Query {
```

Comment:
Discussed with Damian; changing inheritance will muck up the constructor call signatures. The public 3.1 API where this can be accessed is SQLQuery::execute(), and this is documented as returning an SS_Query, so we think that returning a different subclass of SS_Query is fine.